Source/WebCore/Modules/screen-wake-lock/WakeLock.cpp

```diff
@@ -67,9 +67,15 @@ void WakeLock::request(WakeLockType lockType, Ref<DeferredPromise>&& promise)
     // FIXME: The permission check can likely be dropped once the specification gets updated to only
     // require transient activation (https://github.com/w3c/screen-wake-lock/pull/326).
     bool hasTransientActivation = document->domWindow() && document->domWindow()->hasTransientActivation();
-    PermissionController::shared().query(document->clientOrigin(), PermissionDescriptor { PermissionName::ScreenWakeLock }, *document->page(), PermissionQuerySource::Window, [protectedThis = Ref { *this }, document = Ref { *document }, hasTransientActivation, promise = WTFMove(promise), lockType](std::optional<PermissionState> permission) mutable {
-        if (!permission || *permission == PermissionState::Prompt)
-            permission = hasTransientActivation ? PermissionState::Granted : PermissionState::Denied;
+    PermissionController::shared().query(document->clientOrigin(), PermissionDescriptor { PermissionName::ScreenWakeLock }, *document->page(), PermissionQuerySource::Window, [this, protectedThis = Ref { *this }, document = Ref { *document }, hasTransientActivation, promise = WTFMove(promise), lockType](std::optional<PermissionState> permission) mutable {
+        if (!permission || *permission == PermissionState::Prompt) {
+            if (hasTransientActivation || m_wasPreviouslyAuthorizedDueToTransientActivation) {
```

Comment:
Do we need m_wasPreviouslyAuthorizedDueToTransientActivation? can't we check if window->hasStickyActivation?

Reply:
Wouldn't your proposal be a LOT more permissive?
Basically, if I have ever interacted with the document, then the JS will be able to acquire screen wake lock whenever?

I was trying to maintain our current behavior of requiring a transient activation for the initial request.

Reply:
That's true, at the same time. I'm trying to see if we can avoid inventing a "new(ish)" user activation model... but we might need one here 🤔

Reply:
I think there are very few things that depend on sticky, so it might be worth doing. I'll look into that from the HTML side.